cura/Settings/MachineManager.py

```diff
@@ -1314,14 +1314,14 @@ def updateMaterialWithVariant(self, position: Optional[str] = None) -> None:
         """
         if self._global_container_stack is None:
             return
-        if position is None:
-            position_list = [str(position) for position in range(len(self._global_container_stack.extruderList))]
-        else:
+        if position:
             position_list = [position]
+        else:
+            position_list = [str(position) for position, _ in enumerate(self._global_container_stack.extruderList)]
 
-        for position_item in position_list:
+        for position_ in position_list:
```

Comment:
```suggestion
        for position_item in position_list:
```

I think it would be better to change this back to position\_item. Adding the "\_" is sidestepping shadowing the function parameter position but is equally confusing.